device/MatrixMultiplication.cpp

```diff
@@ -308,7 +313,7 @@ void WriteCInner<1>(hlslib::Stream<PackedFloat> &from_kernel, DramLine *const me
         from_kernel.Pop().UnpackFlits(num);
         const bool in_bounds = (n0 * kTileSizeN + n1 < size_n) && (m0 * kTileSizeM + m1 < size_m);
         if (in_bounds) {
-            mem[(n0 * kTileSizeN + n1) * size_m + m0 * kTileSizeM + m1] = num[0];
+            mem[((n0 * kTileSizeN + n1) + (m0 * kTileSizeM + m1) * size_n) * kLinesPerNumber] = num[0];
```

Comment:
```suggestion
            mem[(m0 * kTileSizeM + m1) * size_n + n0 * kTileSizeN + n1] = num[0];
```

Reply:
Wait, this is not good -- C has now also become strided access, so we access _two_ arrays strided, and only one sequentially. I'm afraid we have to complete change the iteration space of the computation for this, not just the memory indices